src/Components/GetWeatherData.js

```diff
@@ -0,0 +1,74 @@
+// ***********************
+// *** Modules & Hooks ***
+// ***********************
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { citiesList } from "../modules/citiesList";
+
+// The function below (getCityID) takes in a city name as a string (i.e. "Halifax", "Toronto", etc) and returns the unique ID for that city that we need for the API call.
+const getCityID = (city) => {
+  const findCityID = citiesList.find(e => e.city === city);
+  return findCityID.id;
+}
+
+// API key and API Url for the axios call
+const apiKey = "oU28MIP77GV30LbR2diTL2ACEcTuAWoZ";
+const baseURL = "https://dataservice.accuweather.com/currentconditions/v1/";
+
+// API call to get weather data
+const GetWeatherData = (props) => {
+  const [weather, setWeather] = useState([]);
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(true);
+
+// find ID of city for API search, add to searchURL
+  const ID = getCityID(props.cityName);
+  const searchURL = `${baseURL}${ID}`; 
+  
+// API axios call inside a useEffect hook, with 1 dependency. This will run anytime the city name changes (props.cityName).
+  useEffect(() => {
```

Comment:
are you using use Effect inside a function?  consider putting it outside unless you want it to make the API call on an event in which case you wouldn't need to use useEffect.